lib/SILOptimizer/Analysis/DifferentiableActivityAnalysis.cpp

```diff
@@ -421,3 +421,32 @@ Activity DifferentiableActivityInfo::getActivity(
     activity |= ActivityFlags::Useful;
   return activity;
 }
+
+void DifferentiableActivityInfo::dump(SILValue value,
+                                      const SILAutoDiffIndices &indices,
+                                      llvm::raw_ostream &s) const {
+  s << '[';
+  auto activity = getActivity(value, indices);
+  switch (activity.toRaw()) {
+  case 0: s << "NONE"; break;
+  case (unsigned)ActivityFlags::Varied: s << "VARIED"; break;
+  case (unsigned)ActivityFlags::Useful: s << "USEFUL"; break;
+  case (unsigned)ActivityFlags::Active: s << "ACTIVE"; break;
+  }
+  s << "] " << value;
+}
+
+void DifferentiableActivityInfo::dump(SILFunction &fn,
```

Comment:
The function argument is unnecessary. `getFunction()` gives you the function.

Reply:
Good catch. Done!